src/vivarium_testing_utils/automated_validation/interface.py

```diff
@@ -399,3 +405,31 @@ def format_ui_data_index(data: pd.DataFrame, comparison_key: str) -> pd.DataFram
         sorted_index = ordered_cols + extra_idx_cols
         sorted = data.reorder_levels(sorted_index).sort_index()
         return add_comparison_metadata_levels(sorted, comparison_key)
+
+    def add_new_measure(self, measure_key: str, measure_class: type[Measure]) -> None:
+        """Add a new measure class to the context's measure mapper.
+        
+        Parameters
+        ----------
+        measure_key
+            The measure key in format 'entity_type.entity.measure_key' or 'entity_type.measure_key'.
+        measure_class
+            The class implementing the measure.
+        """
+        
+        parts = measure_key.split(".")
+        if len(parts) not in (2, 3):
```

Comment:
I think we should encode the structure of the measure key mapping in only one place. That should either be here in the interface or in the measures module. Currently, we've spread this information across both modules in a way that I don't think is strictly necessary.

Reply:
I am not sure the intended change here. We do have a method here and in measures module that do something with different parts of a measure key but they are doing different things. I could add a helper method that just splits the key but I don't think that is what you mean. I think if we were to contain all of this in one module we would have to move get_measure_from_key from measures.py to interface and have that method also handle updating the defaultdict on ValidationContext.